Add mdn_test.py

harry-fuyu opened the pull request:
I've added a test for mdn for data with different number of features. Are there other tests that I should add?

harry-fuyu committed:
added test workflow for pyknos in test_mdn.py

harry-fuyu committed:
added test workflow for pyknos in test_mdn.py

harry-fuyu committed:
Create mdn_test.py

harry-fuyu committed:
Set up automatic testing process. Create mdn_test.py

harry-fuyu committed:
Create __init__.py

alvorithm commented on test/mdn_test.py:
```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
```
For the upcoming release we need to remove this import. It was an advisable optimization but only if you are running Python 3.7 or superior. Since the upcoming release will also accept Python 3.6, so that `sbi` can be used e.g. under Google Colab, this line must be deleted as it would produce an error when using Python 3.6.

alvorithm commented on test/mdn_test.py:
```diff
@@ -0,0 +1,55 @@
+            hidden_features: int=50, 
+            num_components: int=10) -> None:
+
+    theta = torch.rand(3, dim)
```
I am not sure why we are using here theta instead of `x`. The tests should make sense in a general conditional-density estimation setting, and not only in a sbi context.

alvorithm commented on test/mdn_test.py:
```diff
@@ -0,0 +1,55 @@
+    likelihood_cov = eye(dim)
+    context =  linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    x_numel = theta[0].numel()
```
This is confusing - either x or theta (I think x is better here).
Also -- if `dim` is enforced to be a scalar, as we discussed, `x_numel == dim` so there's no need to calculate it?

alvorithm commented on test/mdn_test.py:
```diff
@@ -0,0 +1,55 @@
+    
+    distribution = MultivariateGaussianMDN(
+        features=x_numel,
+        context_features=y_numel,
```
We would rather want the nomenclature `p(y|x)`, i.e. given a supervised dataset `(x,y)_{:n}` we estimate the conditional density `p(y|x)`.

alvorithm reviewed (changes requested):
Let us discuss the naming of the variable before integrating this.

harry-fuyu committed:
updated variable names

harry-fuyu committed:
updated variable names

harry-fuyu committed:
update formatting

harry-fuyu committed:
Create test.yml

harry-fuyu committed:
Update test.yml

harry-fuyu committed:
Update test.yml

harry-fuyu committed:
Update test.yml

harry-fuyu committed:
Update test.yml

Commit:
pinned versions for torch, python and nflows

Commit:
pinned versions for torch, python and nflows

Commit:
parent 20d57506d8c5dbfd59024b193bae3b239698d1f1
author Harry Fu Yu <fu_yu@college.harvard.edu> 1597074385 +0800
committer Harry Fu Yu <fu_yu@college.harvard.edu> 1597079772 +0800

add dev pytest-pep8

Commit:
add dev pytest-pep8

harry-fuyu committed:
Merge branch 'master' into testing_branch

Commit:
Merge branch 'testing_branch' of https://github.com/mackelab/pyknos into testing_branch

Commit:
Merge branches to resolve conflict
